### PR TITLE
Update for MLflow 1.27 release

### PR DESCRIPTION
**Pull request by jinzhang21**
Signed-off-by: Jin Zhang <jin.zhang@databricks.com>

## What changes are proposed in this pull request?

This PR shall be merged after MLflow 1.27.0 is released. I'll perform another round of test before merging it to make sure it works as intended.

## How is this patch tested?
TBD

## Release Notes

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-mlflow
+mlflow>=1.27.0
```
Comment (dbczumar) [inline requirements.txt]: This needs to be `mlflow[pipelines]`, else pipelines won't work

File: README.md [hunk 2]
```diff
@@ -2,10 +2,8 @@
 
 ## Installation instructions
 1. Clone the MLflow Pipelines template repo locally: `git clone https://github.com/mlflow/mlp-regression-template.git`.
-2. Checkout the latest MLflow: `git clone https://github.com/mlflow/mlflow.git`
-3. Install pipeline related requirements in MLflow. `cd mlflow && pip install -e .[pipelines]`
-4. Enter the root of the template: `cd mlp-regression-template`
-5. Install template dendencies: `pip install -r requirements.txt`
+2. Enter the root of the template: `cd mlp-regression-template`
```
Comment (dbczumar) [inline README.md]: Can we tell folks to install `mlflow[pipelines]` separately? I think we should familiarize them with the installation procedure for MLP

File: README.md [hunk 3]
```diff
@@ -15,7 +13,14 @@ To log pipeline runs to a particular MLflow experiment,
...
 [Sync](https://docs.databricks.com/repos.html) this repo and run `notebooks/databricks` on an DBR 11.x cluster with [workspace files support enabled](https://docs.databricks.com/repos.html#work-with-non-notebook-files-in-a-databricks-repo).
 
-**Note** We recommend to open at least **3 browser tabs** to facilitate easier development:
+**Note** When making changes to pipelines on Databricks,
+it is recommended that you either edit files on your local machine and
+use dbx to sync them to Databricks Repos, as demonstrated [here](https://mlflow.org/docs/latest/pipelines.html#usage),
+or edit files in Databricks Repos by opening separate browser tabs
+for each YAML file or Python code module that you wish to modify.
+
+For the later approach,
```
Comment (dbczumar) [inline README.md]:
```suggestion
For the latter approach,
```

**Review by dbczumar — APPROVED** [review]
LGTM with a few small comments. Thanks @jinzhang21 !

**Review by jinzhang21 — COMMENTED** [review]
Thanks, Corey! I'll make sure to test these in a clean environment before merging.

**Comment by dbczumar** [comment]
> Thanks, Corey! I'll make sure to test these in a clean environment before merging.

Sounds good! Can we get this merged tonight?

File: requirements.txt [hunk 4]
```diff
@@ -1,4 +1,4 @@
-mlflow
+mlflow[pipelines]>=1.27.0
 pandas==1.3.*
 pandas-profiling==3.1.*
 scikit-learn==1.0.*
```
Comment (dbczumar) [inline requirements.txt]: @jinzhang21 On second thought, I think these requirements are too strict. `mlflow[pipelines]` requires >=, generally, so users may install MLP and then have to downgrade libraries when using the template. Can we make these reqs looser. IMO, we should unpin dependencies that don’t explicitly need to be pinned, replacing them with lower bounds. Perhaps scikit-learn and pandas should be pinned for ML reproducibility, though by extension we would also want to pin numpy, etc. We should also update databricks.py and jupyter.ipynb to install from the requirements file before running additional cells

**Pull request merged**